Code/Framework/AzToolsFramework/AzToolsFramework/Manipulators/PaintBrushManipulator.cpp

```diff
@@ -159,84 +190,150 @@ namespace AzToolsFramework
+
+                // Loop through every point that's been provided and see if it has a valid paint opacity.
                 for (size_t index = 0; index < points.size(); index++)
                 {
-                    // Any point that falls within our brush stroke is considered valid. Since our brush is a circle, the brush stroke
-                    // ends up being a capsule with a circle on each end, and effectively a box connecting the two circles.
-                    // We can easily check to see if the point falls in the capsule by getting the distance between the point
-                    // and the line segment formed between the center of the start and end circles.
-                    // If this distance is less than the radius of the circle, then it falls within the brush stroke.
-                    // This works equally well whether the point is closest to an endpoint of the segment or a point along the segment.
-                    if (AZ::Geometry2DUtils::ShortestDistanceSqPointSegment(AZ::Vector2(points[index]), previousCenter2D, center2D) <=
-                         manipulatorRadiusSq)
+                    float opacity = 0.0f;
+                    AZ::Vector2 point2D(points[index]);
+
+                    // Loop through each stamp that we're drawing and accumulate the results for this point.
+                    for (auto& brushCenter : brushStamps)
+                    {
+                        // Since each stamp is a circle, we can just compare distance to the center of the circle vs radius.
+                        if (float shortestDistanceSquared = brushCenter.GetDistanceSq(point2D);
+                            shortestDistanceSquared <= manipulatorRadiusSq)
+                        {
+                            // It's a valid point, so calculate the opacity. The per-point opacity for a paint stamp is a combination
+                            // of the hardness falloff and the flow. The flow value gives the overall opacity for each stamp, and the
+                            // hardness falloff gives per-pixel opacity within the stamp.
+                            // For the falloff function, we use a nonlinear falloff that's approximately the same as a squared cosine:
+                            // 2x^3 - 3x^2 + 1
+                            float shortestDistanceNormalized = sqrt(shortestDistanceSquared) / radius;
+                            float hardnessDistance = AZStd::max(shortestDistanceNormalized - hardness, 0.0f) * inverseHardnessReciprocal;
+                            float curHardness = (2.0f * hardnessDistance * hardnessDistance * hardnessDistance) -
+                                (3.0f * hardnessDistance * hardnessDistance) + 1.0f;
+
+                            // For the opacity at this point, combine the opacity from previous stamps with the current hardness and flow.
+                            opacity = AZStd::min(opacity + (curHardness * flow), 1.0f);
```

Comment:
If I understand this correctly, I don't think this is right. This also applies to 'OnPaint()' below (in EditorImageGradientComponentMode.cpp).

This appears to just be adding opacities and clamping at 1.0. So, if you set flow to 0.5 and the hardness was 1.0, any stamp that overlapped the previous would be 0.5 + 0.5 = 1.0, completely opaque. But it should be only 0.75 opaque. The math should be something like
```
float stampOpacity = curHardness * flow;
opacity = opacity + (1.0f - opacity) * stampOpacity;
```

Reply:
Awesome catch, thx! I really thought these were additive in the paint programs, not blended. You're 100% correct. Changed!